recipes/cairo/all/conanfile.py

```diff
@@ -0,0 +1,256 @@
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+import shutil
+
+
+class CairoConan(ConanFile):
+    name = "cairo"
+    description = "Cairo is a 2D graphics library with support for multiple output devices"
+    topics = ("conan", "cairo", "graphics")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://cairographics.org/"
+    license = ("LGPL-2.1-only", "MPL-1.1")
+    exports_sources = ["patches/*"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+        "fPIC": [True, False],
+        "enable_ft": [True, False],
+        "enable_fc": [True, False],
+        "enable_xlib": [True, False],
+        "enable_xlib_xrender": [True, False],
+        "enable_xcb": [True, False],
+        "enable_glib": [True, False]}
+    default_options = {'shared': False,
+        'fPIC': True,
+        "enable_ft": True,
+        "enable_fc": True,
+        "enable_xlib": True,
+        "enable_xlib_xrender": False,
+        "enable_xcb": True,
+        "enable_glib": True}
+    generators = "pkg_config"
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    _env_build = None
+
+    def config_options(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+            del self.options.enable_fc
+        if self.settings.os != 'Linux':
+            del self.options.enable_xlib
+            del self.options.enable_xlib_xrender
+            del self.options.enable_xcb
+
+    def configure(self):
+        if self._is_msvc:
+            if self.settings.build_type not in ['Debug', 'Release']:
+                raise ConanInvalidConfiguration("MSVC build supports only Debug or Release build type")
+
+    def requirements(self):
+        if self.options.enable_ft:
+            self.requires("freetype/2.10.4")
+        if self.settings.os != "Windows" and self.options.enable_fc:
+            self.requires("fontconfig/2.13.91")
+        if self.settings.os == 'Linux':
+            if self.options.enable_xlib or self.options.enable_xlib_xrender or self.options.enable_xcb:
+                self.requires("xorg/system")
+        if self.options.enable_glib:
+            self.requires("glib/2.67.0")
+        self.requires("zlib/1.2.11")
+        self.requires("pixman/0.40.0")
+        self.requires("libpng/1.6.37")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires('msys2/20190524')
+        self.build_requires("pkgconf/1.7.3")
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == 'Visual Studio'
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
+        os.rename('cairo-%s' % self.version, self._source_subfolder)
+        
+
+    def build(self):
```

Comment:
Little formatting fix
```suggestion
    def source(self):
        tools.get(**self.conan_data["sources"][self.version])
        os.rename("cairo-{}".format(self.version), self._source_subfolder)

    def build(self):
```